BLUESPAWN-client/src/hunt/hunts/HuntT1031.cpp

```diff
@@ -0,0 +1,84 @@
+#include "hunt/hunts/HuntT1031.h"
+
+#include "util/filesystem/FileSystem.h"
+#include "util/configurations/Registry.h"
+#include "hunt/RegistryHunt.h"
+#include "util/filesystem/YaraScanner.h"
+#include "util/processes/ProcessUtils.h"
+
+#include "util/log/Log.h"
+
+using namespace Registry;
+
+namespace Hunts {
+
+	HuntT1031::HuntT1031() : Hunt(L"T1031 - Modify Existing Service") {
+		dwSupportedScans = (DWORD) Aggressiveness::Cursory;
+		dwCategoriesAffected = (DWORD) Category::Configurations | (DWORD) Category::Files;
+		dwSourcesInvolved = (DWORD) DataSource::Registry | (DWORD) DataSource::Services;
+		dwTacticsUsed = (DWORD) Tactic::Persistence;
+	}
+
+	int HuntT1031::ScanCursory(const Scope& scope, Reaction reaction){
+		LOG_INFO("Hunting for " << name << " at level Cursory");
+		reaction.BeginHunt(GET_INFO());
+
+		int detections = 0;
+
+		std::vector<RegistryValue> dnsServerPlugins{ CheckValues(HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Services\\DNS\\Parameters", {
```

Comment:
Is this the only well known service that uses a separate dll?

Reply:
nope, but it's a good enough start. Addressing other, higher priority things for the moment